### ISSUE TITLE
Add New Reusable Block interface displaying blank page

### ISSUE THREAD

**Issue by willdelphia** [WordPress/gutenberg#12324]
**Describe the bug**

Add new (reusable) block feature seems to be broken. When accessing through link on Manage All Reusable Blocks page or directly at `.../wp-admin/post-new.php?post_type=wp_block`

Error in console: 

`api-fetch.min.js?ver=2.2.5:1           GET http://gutentest.local/wp-json/wp/v2/blocks/43?context=edit&_locale=user 403 (Forbidden)`

`redux-routine.min.js?ver=3.0.3:1          Uncaught (in promise) Error: [object Object]`

**To Reproduce**

1. Go to any post or page
2. Click on Three Dots (More tools and options) icon
3. Click on "Manage All Reusable Blocks" (..../wp-admin/edit.php?post_type=wp_block)
4. Click add new (.../wp-admin/post-new.php?post_type=wp_block)
5. See completely blank page and errors in console. 

**Expected behavior**
I would expect a new reusable block creation interface. 

**Desktop (please complete the following information):**
 - OS: MAC OS Mojave
 - Browser: Chrome Version 70.0.3538.102 (Official Build) (64-bit) 

**Additional context**
- Wordpress beta 5.0-RC1-43944

**Comment by designsimply** [comment]
I tested with WordPress 4.9.8 and Gutenberg 4.5.1 using Firefox 63.0.3 on macOS 10.13.6 and was able to go to Reusable Blocks > Add New without any problems.

![12324](https://user-images.githubusercontent.com/1119271/49057016-2662fd00-f1bb-11e8-90ff-9adb9e2d4d22.gif)
<sup>Seen at https://tan-otter.jurassic.ninja/wp-admin/edit.php?post_type=wp_block running WordPress 4.9.8 and Gutenberg 4.5.1 using Firefox 63.0.3 on macOS 10.13.6.</sup>

Testing note: try testing with Chrome and then try testing with WordPress 5.0-RC1 to see if it is a problem happening only with one of those constraints.

**Comment by mrwweb** [comment]
This certainly deserves more testing, but I couldn't replicate this bug or find that error in the console.

Win 10, Firefox 63.0.3, WordPress 5.0-RC1-43944, Twenty Nineteen, no plugins.

**Comment by willdelphia** [comment]
I think it's worth mentioning that I am running the wordpress instance on a "Local By Flywheel" VM. 
 (https://local.getflywheel.com/)

Also here's A complete copy of the console output, in case this helps illuminate the cause. 

```
JQMIGRATE: Migrate is installed, version 1.4.1
[Violation] Forced reflow while executing JavaScript took 38ms
api-fetch.min.js?ver=2.2.5:1 GET http://bugtest2.local/wp-json/wp/v2/blocks/13?context=edit&_locale=user 403 (Forbidden)
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
tryCatch @ wp-polyfill.min.js?ver=7.0.0:3
invoke @ wp-polyfill.min.js?ver=7.0.0:3
t.(anonymous function) @ wp-polyfill.min.js?ver=7.0.0:3
n @ api-fetch.min.js?ver=2.2.5:1
c @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
g @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
i @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
m @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
i @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
_ @ api-fetch.min.js?ver=2.2.5:1
API_FETCH @ core-data.min.js?ver=2.0.14:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
n.any @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
n.any @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
n.any @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
Promise.then (async)
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
t @ redux-routine.min.js?ver=3.0.3:1
n.iterator @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
n.any @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
t @ redux-routine.min.js?ver=3.0.3:1
n.iterator @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
t @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ data.min.js?ver=4.0.1:1
tryCatch @ wp-polyfill.min.js?ver=7.0.0:3
invoke @ wp-polyfill.min.js?ver=7.0.0:3
t.(anonymous function) @ wp-polyfill.min.js?ver=7.0.0:3
n @ data.min.js?ver=4.0.1:1
c @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
w @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
fulfill @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
tryCatch @ wp-polyfill.min.js?ver=7.0.0:3
invoke @ wp-polyfill.min.js?ver=7.0.0:3
t.(anonymous function) @ wp-polyfill.min.js?ver=7.0.0:3
n @ data.min.js?ver=4.0.1:1
c @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
a @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
(anonymous) @ edit-post.min.js?ver=3.1.2:12
n @ data.min.js?ver=4.0.1:1
r @ data.min.js?ver=4.0.1:1
zf @ react-dom.min.js?ver=16.6.3:69
Mf @ react-dom.min.js?ver=16.6.3:87
ph @ react-dom.min.js?ver=16.6.3:98
eg @ react-dom.min.js?ver=16.6.3:125
fg @ react-dom.min.js?ver=16.6.3:126
wc @ react-dom.min.js?ver=16.6.3:138
fa @ react-dom.min.js?ver=16.6.3:137
gg @ react-dom.min.js?ver=16.6.3:135
Ca @ react-dom.min.js?ver=16.6.3:133
og @ react-dom.min.js?ver=16.6.3:150
Rd @ react-dom.min.js?ver=16.6.3:150
bb.render @ react-dom.min.js?ver=16.6.3:200
(anonymous) @ react-dom.min.js?ver=16.6.3:153
mg @ react-dom.min.js?ver=16.6.3:148
Bc @ react-dom.min.js?ver=16.6.3:153
render @ react-dom.min.js?ver=16.6.3:202
Zn @ edit-post.min.js?ver=3.1.2:12
(anonymous) @ post-new.php?post_type=wp_block:1696
api-fetch.min.js?ver=2.2.5:1 Fetch failed loading: GET "http://bugtest2.local/wp-json/wp/v2/blocks/13?context=edit&_locale=user".
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
tryCatch @ wp-polyfill.min.js?ver=7.0.0:3
invoke @ wp-polyfill.min.js?ver=7.0.0:3
t.(anonymous function) @ wp-polyfill.min.js?ver=7.0.0:3
n @ api-fetch.min.js?ver=2.2.5:1
c @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
g @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
i @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
m @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
i @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
(anonymous) @ api-fetch.min.js?ver=2.2.5:1
_ @ api-fetch.min.js?ver=2.2.5:1
API_FETCH @ core-data.min.js?ver=2.0.14:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
n.any @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
n.any @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
n.any @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
Promise.then (async)
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
t @ redux-routine.min.js?ver=3.0.3:1
n.iterator @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
n.any @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
t @ redux-routine.min.js?ver=3.0.3:1
n.iterator @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
r @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
t @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ data.min.js?ver=4.0.1:1
tryCatch @ wp-polyfill.min.js?ver=7.0.0:3
invoke @ wp-polyfill.min.js?ver=7.0.0:3
t.(anonymous function) @ wp-polyfill.min.js?ver=7.0.0:3
n @ data.min.js?ver=4.0.1:1
c @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
w @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
fulfill @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
tryCatch @ wp-polyfill.min.js?ver=7.0.0:3
invoke @ wp-polyfill.min.js?ver=7.0.0:3
t.(anonymous function) @ wp-polyfill.min.js?ver=7.0.0:3
n @ data.min.js?ver=4.0.1:1
c @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
a @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
(anonymous) @ edit-post.min.js?ver=3.1.2:12
n @ data.min.js?ver=4.0.1:1
r @ data.min.js?ver=4.0.1:1
zf @ react-dom.min.js?ver=16.6.3:69
Mf @ react-dom.min.js?ver=16.6.3:87
ph @ react-dom.min.js?ver=16.6.3:98
eg @ react-dom.min.js?ver=16.6.3:125
fg @ react-dom.min.js?ver=16.6.3:126
wc @ react-dom.min.js?ver=16.6.3:138
fa @ react-dom.min.js?ver=16.6.3:137
gg @ react-dom.min.js?ver=16.6.3:135
Ca @ react-dom.min.js?ver=16.6.3:133
og @ react-dom.min.js?ver=16.6.3:150
Rd @ react-dom.min.js?ver=16.6.3:150
bb.render @ react-dom.min.js?ver=16.6.3:200
(anonymous) @ react-dom.min.js?ver=16.6.3:153
mg @ react-dom.min.js?ver=16.6.3:148
Bc @ react-dom.min.js?ver=16.6.3:153
render @ react-dom.min.js?ver=16.6.3:202
Zn @ edit-post.min.js?ver=3.1.2:12
(anonymous) @ post-new.php?post_type=wp_block:1696
redux-routine.min.js?ver=3.0.3:1 Uncaught (in promise) Error: [object Object]
    at redux-routine.min.js?ver=3.0.3:1
    at redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
(anonymous) @ redux-routine.min.js?ver=3.0.3:1
Promise.then (async)
n @ data.min.js?ver=4.0.1:1
c @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
a @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
(anonymous) @ data.min.js?ver=4.0.1:1
(anonymous) @ edit-post.min.js?ver=3.1.2:12
n @ data.min.js?ver=4.0.1:1
r @ data.min.js?ver=4.0.1:1
zf @ react-dom.min.js?ver=16.6.3:69
Mf @ react-dom.min.js?ver=16.6.3:87
ph @ react-dom.min.js?ver=16.6.3:98
eg @ react-dom.min.js?ver=16.6.3:125
fg @ react-dom.min.js?ver=16.6.3:126
wc @ react-dom.min.js?ver=16.6.3:138
fa @ react-dom.min.js?ver=16.6.3:137
gg @ react-dom.min.js?ver=16.6.3:135
Ca @ react-dom.min.js?ver=16.6.3:133
og @ react-dom.min.js?ver=16.6.3:150
Rd @ react-dom.min.js?ver=16.6.3:150
bb.render @ react-dom.min.js?ver=16.6.3:200
(anonymous) @ react-dom.min.js?ver=16.6.3:153
mg @ react-dom.min.js?ver=16.6.3:148
Bc @ react-dom.min.js?ver=16.6.3:153
render @ react-dom.min.js?ver=16.6.3:202
Zn @ edit-post.min.js?ver=3.1.2:12
(anonymous) @ post-new.php?post_type=wp_block:1696
```

**Comment by mrwweb** [comment]
My tests were also with Local.

**Comment by willdelphia** [comment]
Update: 

If I spin a completely new wordpress install with local and then add the Beta testing plugin and update to "Bleeding Edge Nightly" (In this case 5.0-RC1-43944) I get the error behavior. 

However if after taking the steps above but I also install the plugin called "Gutenberg" the error stops and I am able to see the appropriate page at `wp-admin/post-new.php?post_type=wp_block`

**Comment by willdelphia** [comment]
Same results when I tested on a new Amazon Lightsail instance.  At first broken, then I added the 'Gutenberg' plugin and error stopped.

**Comment by designsimply** [comment]
I tested with WordPress 5.0-RC1-43945 on Chrome 70.0.3538.110 on macOS 10.13.6 and still cannot replicate. Host is Bluehost.

I noticed a 403 error in the console output you posted:

> GET http://bugtest2.local/wp-json/wp/v2/blocks/13?context=edit&_locale=user 403 (Forbidden)

Can you verify the API isn't being blocked for some reason?

**Comment by willdelphia** [comment]
When I go directly to that url I see json:

{"code":"rest_forbidden_context","message":"Sorry, you are not allowed to edit this post.","data":{"status":401}}

**Comment by designsimply** [comment]
🤔 ok, the API needs to be accessible. I'm not 100% sure whether we're on the right track, but do you have a local firewall or security rules or a security plugin in your local setup even which might be the source of the trouble in this case?

I re-tested with a local install running WordPress 4.9.8 and Gutenberg 4.6 `master` @ `4178f2da4` and I was able to go to `/wp-admin/edit.php?post_type=wp_block` and click "Add New" to create and publish a reusable block using my local dev setup.

**Comment by dhirenpatel22** [comment]
> **Describe the bug**
> 
> Add new (reusable) block feature seems to be broken. When accessing through link on Manage All Reusable Blocks page or directly at `.../wp-admin/post-new.php?post_type=wp_block`
> 
> Error in console:
> 
> `api-fetch.min.js?ver=2.2.5:1 GET http://gutentest.local/wp-json/wp/v2/blocks/43?context=edit&_locale=user 403 (Forbidden)`
> 
> `redux-routine.min.js?ver=3.0.3:1 Uncaught (in promise) Error: [object Object]`
> 
> **To Reproduce**
> Steps to reproduce the behavior:
> 
> 1. Go to any post or page
> 2. Click on Three Dots (More tools and options) icon
> 3. Click on "Manage All Reusable Blocks" (..../wp-admin/edit.php?post_type=wp_block)
> 4. Click add new (.../wp-admin/post-new.php?post_type=wp_block)
> 5. See completely blank page and errors in console.
> 
> **Expected behavior**
> I would expect a new reusable block creation interface.
> 
> **Desktop (please complete the following information):**
> 
> * OS: MAC OS Mojave
> * Browser: Chrome Version 70.0.3538.102 (Official Build) (64-bit)
> 
> **Additional context**
> 
> * Wordpress beta 5.0-RC1-43944

I was able to generate the same issue in new fresh WordPress 5.0 setup. When I tried to add 'Reusable Block', it ended up showing me a blank screen.
`/wp-admin/post-new.php?post_type=wp_block`

**Comment by ettoredn** [comment]
I can confirm this bug. It started happening out of the blue some time after updating to WP 5.0 and using Gutenberg. In my situation, I have the "Classic Editor" plugin installed and when trying to edit _any_ post/page with Gutenberg, the editor just doesn't show (blank page). Unable to replicate the issue on my local installation so far.

**Comment by vikaskhunteta** [comment]
I got the same issue on Windows 7 x64, PHP 7.2.12, MySQL 5.6.24, is I am supposed to add 'Gutenberg' plugin?

**Comment by M1k3xUr** [comment]
I have a website already for testing themes and plugins and updated to 5.0. Testing Gutenberg, I get the "update failed" but, it's already published. Not sure what is the problem and the error comes from tinymce. Here are the view and the result.
![untitled-1](https://user-images.githubusercontent.com/16705269/49822463-38ac7180-fd4b-11e8-9195-06aaf9a7f525.jpg)

**Comment by designsimply** [comment]
@M1k3xUr your issue looks unrelated to the rest of the thread here. Based on the error, it appears to be related to this TinyMCE bug https://github.com/tinymce/tinymce/issues/2194. Please try testing with all plugins turned of and using the default theme and again with your custom theme to verify the theme is truly the source of the problem. From there you can either wait for an update from TinyMCE or try adjusting your code to work around the problem defined in the TinyMCE issue.

**Comment by jarekherisz** [comment]
I have the same issue, but when i installed plugin https://pl.wordpress.org/plugins/gutenberg/ it start work correct. Even I disabled or remove this plugin, it still works. Then it not nginix problem but some wordpress. Something what is changed by installing plugin https://pl.wordpress.org/plugins/gutenberg/

**Comment by meijerwynand** [comment]
Greetings, 

Did a vanilla install of WordPress 5.0.2 via Softalicious installer on Cpanel. 

Once WP was up, I installed and activated the recommended plugins.

Edited the "Sample Page"
- Able to "Add to reusable block"

Open "Manage All Reusable Blocks"
- Unable to create "New" - only provides a blank page
- If saved block is edited and saved as 'draft'/unpublished the block can not be selected to edit and the export json function does not work.

Edited the "Sample Page" - reopening
- Reusable blocks options are not available when you select the (+) to add blocks


Same error as reported [here](#issuecomment-442586429) in dev tools

**Comment by heydey** [comment]
Hello, 
Similar experience with vanilla 5.0.2 installation on MAMP/Macos/Mojave. Click to add new block:
{"code":"rest_forbidden_context","message":"Sorry, you are not allowed to edit this post.","data":{"status":401}}

**Comment by meijerwynand** [comment]
### Solved my issue

@heydey 

After the vanilla install mentioned [above](#issuecomment-451089215) I installed the [Gutenberg Plugin](https://wordpress.org/plugins/gutenberg/) and I was able to make use of the reusable blocks.

**Comment by DenMette** [comment]
So, I have the same issue.
I solved my issue like @meijerwynand explained, so maybe take a look what you missed ... in the plugin.

**Comment by kjnedrud** [comment]
I've been getting the same issue and did some digging. I noticed that the request URL contains a post ID which is interesting because this only happens when trying to create a *new* reusable block (editing existing blocks works fine). So I checked my database for the post ID in question and there is a post with that ID with the status `auto-draft`. Additionally, when I checked the wp_postmeta table for the same post ID, I found an `_edit_lock` row. So it appears that this is caused by WordPress creating an auto draft for a new wp_block post and locking it for editing. I'll keep looking into this further but hopefully this sheds a bit more light on the situation.

**Comment by designsimply** [comment]
I re-tested today using WP 5.0.4-alpha-44523 and also with WP 5.0.3 using test sites created using https://jurassic.ninja/ and I can still repeat the problem but I found something interesting… if I activate the plugin _at any point_ then it starts working normally even if I subsequently deactivate and delete the plugin! ([1m55s](https://cloudup.com/cl1Vik1bsKU))

Steps to reproduce:

1. Go to https://jurassic.ninja to create a new test site.
1. Delete all plugins.
1. Go to Posts > Add New > More > Manage All Reusable Blocks.
1. Click the "Add New" button and note the blank screen and the error in the console.
1. Go to Plugins > Add New and search for Gutenberg.
1. Install and activate Gutenberg then deactivate and delete it.
1. Go to Posts > Add New > More > Manage All Reusable Blocks.
1. Click the "Add New" button and note that it works normally.

<strong>Result:</strong> even if the "Add New" button on the "Manage All Reusable Blocks" wasn't working before, it starts working after the Gutenberg plugin is activated even if it is subsequently deleted. What a strange 🐛!!

<sup>Seen at https://bad-emu-474.jurassic.ninja/wp-admin/post-new.php?post_type=wp_block tested with WordPress 5.0.3 and with/without Gutenberg 4.8.0 using jurassic.ninja and checked from Firefox 64.0.2 on macOS 10.13.6.</sup>

**Comment by desrosj** [comment]
So worked through this a bit today. I posted a [lengthy description on the Trac ticket ](https://core.trac.wordpress.org/ticket/45373#comment:9). 

In summary: the REST API request to the blocks endpoint is returning a 403 error. I believe that this is causing the blank screen error that is being stumbled on. This should be fixed here. A REST API error should not cause a white screen for the user when editing a reusable block. At a minimum, an error message should be displayed with a path forward. 

When the Gutenberg plugin is activated, the Add New Block page works and it continues to work even after the plugin is deactivated. This happens because the plugin adds block capabilities to the user roles, which are stored in the database. These are not cleaned up after the plugin is deactivated/removed, so they remain and the functionality continues to work.

This second issue brings up a bug that could probably be fixed in both locations (Core and the Gutenberg Plugin) so that there is consistency moving forward. The bug in core causing the 403 permissions issue should be fixed, and then, potentially, the plugin should have its capabilities updated to match what is currently in core.

**Comment by desrosj** [comment]
I'm adding this to 4.9 because IMO this is a bug that should be fixed in 5.1.

**Comment by youknowriad** [comment]
Agreed, this seems like a critical issue that should be backported to 5.1.

Just a small not on milestones and workflow, we use Gutenberg versions (4.9, 5.0...) for PRs and WordPress versions for issues.

**Comment by aduth** [comment]
There are a few contributing issues here:

- [Core assigns `is-fullscreen-mode` to the block editor screen](https://github.com/WordPress/WordPress/blob/9fc53b8461ed4419f1fdcb1b85075e22dd52adff/wp-admin/admin-header.php#L191) and relies on the editor to [unset this class](https://github.com/WordPress/gutenberg/blob/0c1f125ed5d2b77ad10673a6bb6970d7b66d98be/packages/edit-post/src/components/fullscreen-mode/index.js#L34-L41) if the user is not using the fullscreen editing option. Thus, if the editor fails to load for any reason, the admin header/sidebar will not be visible, despite the markup being present.
- The editor [does not render anything until the "post" (in this case, the new block) data is available](https://github.com/WordPress/gutenberg/blob/0c1f125ed5d2b77ad10673a6bb6970d7b66d98be/packages/edit-post/src/editor.js#L24-L26). Since this is not accessible to the user by their capabilities, it will never render. This is why the editor gets stuck in fullscreen mode.
- As mentioned by @desrosj, Gutenberg [explicitly adds the block permissions](https://github.com/WordPress/gutenberg/blob/0c1f125ed5d2b77ad10673a6bb6970d7b66d98be/lib/register.php#L489-L533), while in core these are [remapped to an equivalent post capability](https://github.com/WordPress/WordPress/blob/9fc53b8461ed4419f1fdcb1b85075e22dd52adff/wp-includes/capabilities.php#L585-L600)
   - These should be aligned. Specifically, this code ought to be removed from Gutenberg as part of a bump to require 5.0 as the minimum supported WordPress version.

To me then, the actionable items are:

- Gutenberg be made more tolerant to failed initialization
   - Render `<FullscreenMode />` earlier to cancel fullscreen mode
      - Alternatively, avoid this being assigned by default in core
   - Avoid [uncaught thrown error in failed request from data resolver](https://github.com/WordPress/gutenberg/blob/0c1f125ed5d2b77ad10673a6bb6970d7b66d98be/packages/core-data/src/resolvers.js#L45)
      - With fallback messaging about failed initialization?
- Core permissions need to be corrected for handling block editing
   - The capabilities mapping is hard to follow, but it seems to [fail](https://github.com/WordPress/WordPress/blob/abcb59f86055139a4cce02aff2c013ee7cd17ff4/wp-includes/class-wp-user.php#L779) at a point of considering the user to have an `edit_blocks` capability
   - **In my testing, the most direct fix is to add a new `capabilities` entry `'edit_posts' => 'edit_posts'` to the [core registration of the `wp_block` type](https://github.com/WordPress/WordPress/blob/abcb59f86055139a4cce02aff2c013ee7cd17ff4/wp-includes/post.php#L293-L298).**

There's still an open question as to whether this ought to have been handled already with the [equivalent capabilities mapping](https://github.com/WordPress/WordPress/blob/abcb59f86055139a4cce02aff2c013ee7cd17ff4/wp-includes/capabilities.php#L585-L600). I think the issue here is that the capabilities mapping doesn't occur recursively. The triggering condition [occurs on the singular `'edit_block'` capability](https://github.com/WordPress/WordPress/blob/abcb59f86055139a4cce02aff2c013ee7cd17ff4/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L1384). Given the `wp_block` type is [registered with `map_meta_cap`](https://github.com/WordPress/WordPress/blob/abcb59f86055139a4cce02aff2c013ee7cd17ff4/wp-includes/post.php#L191), this is then [mapped to the `edit_post` capability](https://github.com/WordPress/WordPress/blob/abcb59f86055139a4cce02aff2c013ee7cd17ff4/wp-includes/capabilities.php#L580-L583) and calls again to `map_meta_cap`. This in-turn [maps back to `'edit_blocks'`](https://github.com/WordPress/WordPress/blob/abcb59f86055139a4cce02aff2c013ee7cd17ff4/wp-includes/capabilities.php#L182-L183) which is not an assigned capability for the user. My thinking is that either the `wp_block` type should not be registered with `map_meta_cap`, or the post-equivalent mapping should be reflected through the post type registration's `capability` property than in `map_meta_cap`, but I'm not too certain on how the current implementation was approached (cc @pento re: [WP#43804](https://core.trac.wordpress.org/changeset/43804)).

**Comment by condensed-io** [comment]
hope this helps someone because I ended up on this page when looking for the useless 'publishing failed' error message.  For me it was because my session had timed out and all I had to do was refresh the page and retry publishing.  Had this happen to clients recently too, not sure if the latest version of wordpress introduced something that's logging people out?

**Comment by designsimply** [comment]
> not sure if the latest version of wordpress introduced something that's logging people out?

@condensed-io I don't think so. Also, the session timing out and being logged out are slightly different cases. Check out the issue at https://github.com/WordPress/gutenberg/issues/9260 which sounds to me like it's closer to what you've described.
